network/framework/src/application/interface.rs

```diff
@@ -38,8 +38,11 @@ pub trait NetworkClientInterface<Message: NetworkMessageTrait>: Clone + Send + S
 
     /// Requests that the network connection for the specified peer
     /// is disconnected.
-    // TODO: support disconnect reasons.
-    async fn disconnect_from_peer(&self, _peer: PeerNetworkId) -> Result<(), Error>;
+    async fn disconnect_from_peer(
+        &self,
+        _peer: PeerNetworkId,
+        _reason: String,
```

Comment:
It's probably cleaner to make this an enum, e.g., the first 2 entries could be something like `FailedHealthCheckPing` and `FailedPeerMonitoringPing`

Reply:
Sounds good. There was already a `DisconnectReason` enum in `network > peer/mod.rs` so I will add variants there